src/main/java/com/splunk/kafka/connect/SplunkSinkTask.java

```diff
@@ -250,39 +288,98 @@ private Event createHecEventFrom(final SinkRecord record) {
+        }
+        else {
+            event = createHECEventNonFormatted(record);
         }
 
-        Map<String, String> metas = connectorConfig.topicMetas.get(record.topic());
-        if (metas != null) {
-            event.setIndex(metas.get(SplunkSinkConnectorConfig.INDEX));
-            event.setSourcetype(metas.get(SplunkSinkConnectorConfig.SOURCETYPE));
-            event.setSource(metas.get(SplunkSinkConnectorConfig.SOURCE));
-            event.addFields(connectorConfig.enrichments);
+        if(connectorConfig.headerSupport) {
+            addHeadersToHecEvent(event, record);
         }
 
         if (connectorConfig.trackData) {
-            // for data loss, latency tracking
             Map<String, String> trackMetas = new HashMap<>();
             trackMetas.put("kafka_offset", String.valueOf(record.kafkaOffset()));
             trackMetas.put("kafka_timestamp", String.valueOf(record.timestamp()));
             trackMetas.put("kafka_topic", record.topic());
             trackMetas.put("kafka_partition", String.valueOf(record.kafkaPartition()));
             event.addFields(trackMetas);
         }
-
         event.validate();
 
         return event;
     }
 
+    private Event addHeadersToHecEvent(Event event, SinkRecord record) {
+        //TODO: Will need exception handling around String serialization
+        Headers headers = record.headers();
+        log.debug(headers.toString());
+        if (!headers.isEmpty()) {
+            if (headers.lastWithName(connectorConfig.headerIndex) != null) {
+                log.debug("splunk_index header detected, value is: " + headers.lastWithName(connectorConfig.headerIndex).value().toString());
+                event.setIndex(headers.lastWithName(connectorConfig.headerIndex).value().toString());
+            }
+            if (headers.lastWithName(connectorConfig.headerHost) != null) {
+                log.debug("splunk_host header detected, value is: " + headers.lastWithName(connectorConfig.headerHost).value().toString());
+                event.setHost(headers.lastWithName(connectorConfig.headerHost).value().toString());
+            }
+            if (headers.lastWithName(connectorConfig.headerSource) != null) {
+                log.debug("splunk_source header detected, value is: " + headers.lastWithName(connectorConfig.headerSource).value().toString());
+                event.setSource(headers.lastWithName(connectorConfig.headerSource).value().toString());
+            }
+            if (headers.lastWithName(connectorConfig.headerSourcetype) != null) {
+                log.debug("splunk_sourcetype header detected, value is: " + headers.lastWithName(connectorConfig.headerSourcetype).value().toString());
+                event.setSourcetype(headers.lastWithName(connectorConfig.headerSourcetype).value().toString());
+            }
+
+            if (!connectorConfig.headerCustom.isEmpty()) {
+                String[] customHeaders = connectorConfig.headerCustom.split(",");
```

Comment:
we will need document the custom header format in the code for better maintaince

Reply:
Improved configuration docs and added in-line comments